gax/src/main/java/com/google/api/gax/core/GaxProperties.java

```diff
@@ -57,12 +57,16 @@ public static String getLibraryVersion(Class<?> libraryClass) {
    * method are expected to be cached.
    */
   public static String getLibraryVersion(Class<?> libraryClass, String propertyName) {
-    String version = getLibraryVersion(libraryClass);
-    if (!DEFAULT_VERSION.equals(version)) {
-      return version;
+    String version = null;
+    // Always read GaxProperties' version from the properties file.
+    if (!libraryClass.equals(GaxProperties.class)) {
+      version = getLibraryVersion(libraryClass);
+      if (!DEFAULT_VERSION.equals(version)) {
+        return version;
+      }
     }
 
-    try (InputStream in = libraryClass.getResourceAsStream("/dependencies.properties")) {
```

Comment:
Why is this needed? Most probably this will break execution as a bazel artifact (i.e. when a client library is build with bazel and executed from bazel context). In any case, introducing another properties file will duplicate version information in two places. Plus, the version in `dependencies.properties` is automatically bumped by Yoshi's tools during releases.

Reply:
This is similar to [this PR](https://github.com/googleapis/google-http-java-client/pull/919/files) in google-http-java-client. Note that `dependencies.properties` was never being pulled into the Android JAR. This file is also suboptimal to include since it includes every dependency, which we don't necessarily want gax-external developers to parse. 

On the other hand, `gradle.properties` is a [standard artifact](https://docs.gradle.org/current/userguide/build_environment.html#sec:gradle_configuration_properties) that is expected to be provided in many libraries and works on both Android and plain Java binaries.

Reply:
tl;dr; Please strongly consider just making `dependencies.properties` work in android env.

`google-http-java-client` is not used via bazel, that is why it does not matter for it. Also it does not have `dependencies.properites`.

Note, `dependencies.properties` is used in any scenario (with or without bazel). It is the source of truth for the version.

As for `gradle.properties`, it makes gax-java as java artifact depend on its build system (`gradle`, which is one of many) during runtime. The same artifact can be built by maven, for example, which makes gradle.properties an alien thing there. I guess it is also not even included in bazel artfiact unlike dependencies.properties (https://github.com/googleapis/gax-java/blob/master/gax/BUILD.bazel#L43). If you run `bazel test` for gax-java, does it work (most probably it will not)?

In any case, most importantly, this value is not updated automatically so after the next release it will start reporting wrong version number. There is TODO, but it is not sufficient (after all this PR fixes one relatively small issue, so introducing a future issue with it is not "cost efficient").

Reply:
IMO we shouldn't include all of `dependencies.properties`, so here's a workaround. I'm now generating a new `dependencies.properties` that pulls in `version.gax` at build time. This still works with `bazel test //...`, `./gradlew test`, Android, and plain Java binaries.

We also did not have a `GaxPropertiesTest` before, so I've added one.